crates/nu-parser/src/parse_keywords.rs

```diff
@@ -3366,7 +3367,13 @@ pub fn parse_let(working_set: &mut StateWorkingSet, spans: &[Span]) -> Pipeline
                     let rhs_type = rvalue.ty.clone();
 
                     if let Some(explicit_type) = &explicit_type {
-                        if !type_compatible(explicit_type, &rhs_type) {
+                        // For RHS Type::Any, defer all type checking to runtime.
+                        // This allows the runtime to perform actual compatibility validation
+                        // when the real data is available.
+                        let compatible = matches!(rhs_type, Type::Any)
+                            || type_compatible(explicit_type, &rhs_type);
+
+                        if !compatible {
```

Comment:
I pulled your branch and it seems like this change isn't necessary for the fix to work (tests pass, some examples in REPL work). Did you have any examples where this was needed?